lib/contentful/resource.rb

```diff
@@ -104,11 +104,37 @@ def coerce_value_or_array(value, what = nil)
         nil
       elsif value.is_a? ::Array
         value.map { |v| coerce_or_create_class(v, what) }
+      elsif should_coerce_hash?(value)
+        ::Hash[value.map { |k, v|
+          to_coerce = v.is_a?(Hash) ? v : v.to_s
+          coercion = v.is_a?(Numeric) ? v : coerce_or_create_class(to_coerce, what)
+          [k.to_sym, coercion]
+        }]
       else
         coerce_or_create_class(value, what)
       end
     end
 
+    def should_coerce_hash?(value)
+      value.is_a?(::Hash) &&
+        !self.is_a?(Asset) &&
+        !is_location?(value) &&
+        !is_link?(value) &&
+        !is_image?(value)
```

Comment:
Hm, can we maybe reverse the logic here? Otherwise, I could see us ending up with inconsistencies in which keys are symbols and which are strings very easily if we don't remember amending the list of opt-out cases.